cpp/ql/src/Metrics/Internal/IncludeResolutionStatus.ql

```diff
@@ -0,0 +1,42 @@
+/**
+ * @name Count of includes that failed to be resolved as well as compilation error due to undefined token
+ * @description A count of all failed includes and compilation errors due to undefined tokens.
+ * @kind table
+ * @tags summary telemetry include resolution
+ * @id cpp/telemetry/incliude-resolution-status
+ */
+
+import cpp
+
+/**
+ * A cannot open file error.
+ * Typically this is due to a missing include.
+ */
+class CannotOpenFileError extends CompilerError {
+  CannotOpenFileError() { this.hasTag(["cannot_open_file", "cannot_open_file_reason"]) }
+}
+
+/**
+ * A compiler error happening in a project file.
+ */
+predicate local_compiler_error(CompilerError e) { exists(e.getFile().getRelativePath()) }
+
+/**
+ * A compiler error happening in a project file, where the error is due to a token being undefined.
+ */
+predicate undefined_token_error(CompilerError e) {
+  // The following is a non-exaustive list of tags that indicate that a token
+  // is undefined.
+  e.hasTag("not_a_member") or
+  e.hasTag("undefined_identifier") or
+  e.hasTag("no_matching_function") or
+  e.hasTag("too_many_arguments") or
+  e.hasTag("not_a_class_or_struct_name") or
+  e.hasTag("not_a_class_template")
+}
+
+select count(CannotOpenFileError e) as failed_includes, count( | | 1) as total_files,
```

Comment:
I'm not sure what `count( | | 1) as total_files` is supposed to do. Doesn't this always yield 1?